File: rust/day1/day1_2.rs
use std::fs::File;
use std::io::{BufReader, BufRead};

fn main() {
    let f = File::open("../../inputs/day1/input.txt").unwrap();
	let mut frequency = 0;
	let file = BufReader::new(f);
	for line in file.lines() {
        let number: i32 = line.unwrap().parse().unwrap();
        frequency += number;
    }
	assert_eq!(474 , frequency);
}

File: python/day1/day1_2.py
start_frequency = 0
track_frequency = {}
found_duplicate = False
inputs = []
with open('input.txt') as fp:
    for input in fp:
        inputs.append(int(input))
        
while found_duplicate == False:
    for freq in inputs:
        start_frequency += freq
        if start_frequency in track_frequency:
            found_duplicate = True
            print(start_frequency)
            break
        else:
            track_frequency[start_frequency] = 1


		